doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1189,430 @@ processed in this order:
+
+Project Groups
+==============
+
+Inside ``manifest: projects:``, you can add a project to groups. The ``groups``
+key in this case is a list of group names. Group names are strings.
+
+For example, in this manifest fragment:
+
+.. code-block:: yaml
+
+  manifest:
+    projects:
+      - name: project-1
+        groups:
+          - groupA
+      - name: project-2
+        groups:
+          - groupB
+          - groupC
+      - name: project-3
+
+The projects are in these groups:
+
+- ``project-1``: one group, named ``groupA``
+- ``project-2``: two groups, named ``groupB`` and ``groupC``
+- ``project-3``: no groups
+
+Project group names must not contain commas (,), colons (:), or whitespace.
+
+Group names must not begin with a dash (-), but they may contain dashes
+elsewhere in their names: ``foo-bar`` is a valid group name, but ``-foo-bar``
+is not.
+
+Group names are otherwise arbitrary strings. Group names are case sensitive.
+
+As a restriction, no project may use both ``import:`` and ``groups:``. (This
+avoids some edge cases whose semantics are difficult to specify.)
+
+Active and Inactive Projects
+============================
+
+You can allow or block project groups in both your manifest file and
+:ref:`west-config`.
+
+All projects are *active* by default. A project is *inactive* if it has groups,
+all of which are blocked. This is the only way to make a project inactive:
+projects with no groups are active, as are projects with one or more allowed
+groups.
+
+Within the manifest file, the top level ``manifest: groups:`` value can be used
+to block project groups. To block a group, prefix its name with a dash (-).
+
+For example, in this manifest fragment:
+
+.. code-block:: yaml
+
+   manifest:
+     groups: [-groupA,-groupB]
+
+The groups named ``groupA`` and ``groupB`` are blocked.
+
+Groups are otherwise allowed by default.
+
+This ``groups`` list is also just an ordinary YAML list, so you could have also
+written this fragment like this:
+
+.. code-block:: yaml
+
+   manifest:
+     groups:
+       - -groupA
+       - -groupB
+
+However, this syntax is harder to read and therefore discouraged.
+
+Only the **top level manifest file's** ``manifest: groups:`` value has any
+effect. The ``manifest: groups:`` values in any :ref:`imported manifests
+<west-manifest-import>` are ignored.
+
+In addition to the manifest file, you can control the project group allowlist
+and blocklist using the ``manifest.groups`` configuration option. This option
+is a comma-separated list of groups. To allow a group, add its name to the
+list. To block a group, add its name prefixed by a dash (``-``). For example,
+setting ``manifest.groups`` to ``groupA,-groupB`` allows ``groupA``, and blocks
+``groupB``.
+
+The value of the configuration option overrides any data in the manifest file.
+You can think of this as if the ``manifest.groups`` configuration option is
+appended to the ``manifest: groups:`` list from YAML, with "last entry wins"
+semantics.
+
```

Comment:
Maybe mention here that there's a command-line option to west that has the highest precedence? So that users are aware of it when reading about groups.

Reply:
> Maybe mention here that there's a command-line option to west that has the highest precedence? So that users are aware of it when reading about groups.

I will document this in repo-tool.rst with the rest of the ``west update`` details if that's OK.